test/wtr/app/plugins/preview/preview.test.js

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright 2023 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+/* eslint-disable no-unused-expressions, import/no-extraneous-dependencies */
+
+// @ts-ignore
+import fetchMock from 'fetch-mock/esm/client.js';
+import sinon from 'sinon';
+import { expect, waitUntil } from '@open-wc/testing';
+import { recursiveQuery } from '../../../test-utils.js';
+import chromeMock from '../../../mocks/chrome.js';
+import { AEMSidekick } from '../../../../../src/extension/app/aem-sidekick.js';
+import { mockFetchEnglishMessagesSuccess } from '../../../mocks/i18n.js';
+import { defaultSidekickConfig } from '../../../fixtures/sidekick-config.js';
+import {
+  mockSharepointEditorDocFetchStatusSuccess,
+  mockFetchConfigJSONNotFound,
+  mockGdriveEditorFetchStatusSuccess,
+  mockSharepointEditorSheetFetchStatusSuccess,
+} from '../../../mocks/helix-admin.js';
+import '../../../../../src/extension/index.js';
+import { appStore } from '../../../../../src/extension/app/store/app.js';
+import { HelixMockContentType, mockEditorAdminEnvironment, restoreEnvironment } from '../../../mocks/environment.js';
+import { EventBus } from '../../../../../src/extension/app/utils/event-bus.js';
+import { EVENTS, MODALS } from '../../../../../src/extension/app/constants.js';
+
+// @ts-ignore
+window.chrome = chromeMock;
+
+describe('Preview plugin', () => {
+  let sidekick;
+  beforeEach(async () => {
+    mockFetchEnglishMessagesSuccess();
+  });
+
+  afterEach(() => {
+    document.body.removeChild(sidekick);
+    fetchMock.reset();
+    restoreEnvironment(document);
+  });
+
+  describe('switching between environments', () => {
+    it('previewing from sharepoint editor - docx', async () => {
+      mockSharepointEditorDocFetchStatusSuccess();
+      mockFetchConfigJSONNotFound();
+      mockEditorAdminEnvironment(document, 'editor');
+      const updatePreviewSpy = sinon.stub(appStore, 'updatePreview').resolves();
+      const tipToast = sinon.stub(appStore, 'showToast').returns();
+
+      sidekick = new AEMSidekick(defaultSidekickConfig);
+      document.body.appendChild(sidekick);
+
+      await waitUntil(() => recursiveQuery(sidekick, 'action-bar-picker'));
+
+      const previewPlugin = recursiveQuery(sidekick, '.edit-preview');
+      expect(previewPlugin.textContent.trim()).to.equal('Preview');
+
+      previewPlugin.dispatchEvent(new Event('click'));
```

Comment:
I think you can just use
```suggestion
      previewPlugin.click();
```

Reply:
Today I learned... Thanks @rofe. I've updated any other tests using `dispatchEvent` as well..